File: MeloRenta/src/paquete2/Contrato.java
package paquete2;

import paquete1.Inmueble;

import java.io.File;
import java.time.LocalDate;
import java.util.Date;

public class Contrato {
    private double id;
    private File clausula;
    private LocalDate fecha_inicio;
    private LocalDate fecha_fin;
    private Inmueble inmueble;
    private Inquilino inquilino;
    private boolean estado;

    public Contrato(double id, File clausula, LocalDate fecha_inicio, LocalDate fecha_fin,
                    Inmueble inmueble, Inquilino inquilino, boolean estado) {
        this.id = id;
        this.clausula = clausula;
        this.fecha_inicio = fecha_inicio;
        this.fecha_fin = fecha_fin;
        this.inmueble = inmueble;
        this.inquilino = inquilino;
        this.estado = estado;
    }

    //getter and setter

    public double getId() {
        return id;
    }

    public void setId(double id) {
        this.id = id;
    }

    public File getClausula() {
        return clausula;
    }

    public void setClausula(File clausula) {
        this.clausula = clausula;
    }

    public LocalDate getFecha_inicio() {
        return fecha_inicio;
    }

    public void setFecha_inicio(LocalDate fecha_inicio) {
        this.fecha_inicio = fecha_inicio;
    }

    public LocalDate getFecha_fin() {
        return fecha_fin;
    }

    public void setFecha_fin(LocalDate fecha_fin) {
        this.fecha_fin = fecha_fin;
    }

    public Inmueble getInmueble() {
        return inmueble;
    }

    public void setInmueble(Inmueble inmueble) {
        this.inmueble = inmueble;
    }

    public Inquilino getInquilino() {
        return inquilino;
    }

    public void setInquilino(Inquilino inquilino) {
        this.inquilino = inquilino;
    }

    public boolean isEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    //fin getter and setter

    

    public void renovar_contrato(){ //Aqui tomamos la fecha de finalización del contrato y le sumamos 30 días.
    	this.fecha_fin.plusDays(30);
    }
    
    
    
    

    public String notificar_cierre_contrato(){
    	if (this.fecha_fin == (LocalDate.now().plusDays(1))) {
    		this.estado= !this.estado;
    		String notificacion ="Se ha cerrado el contrato";
    		return notificacion;
		}
    	String notificacion2= "Aún queda tiempo";
    	return notificacion2;
    	
    }
    
    public void inhabilitarContrato() {
    	if (this.estado = true) {
			this.estado= !this.estado;
    	}
    }


    public String mostrar_contrato() {
        return "Contrato{" +
                "id=" + id +
                ", clausula=" + clausula +
                ", fecha_inicio=" + fecha_inicio +
                ", fecha_fin=" + fecha_fin +
                ", inmueble=" + inmueble +
                ", inquilino=" + inquilino +
                ", estado=" + estado +
                '}';
    }
}

File: MeloRentaPython/gestorAplicacion/gestorAplicacion_contratos/Contrato.py
import datetime as dt

class Contrato():
    lista_contratos = []

    def __init__(self, id, clausula, fechaInicio, fechaFin, inmueble, inquilino, estado):
        self._id = id
        self._clausula = clausula
        self._fechaInicio = dt.datetime.strptime(fechaInicio, '%d/%m/%y')
        self._fechaFin = dt.datetime.strptime(fechaFin, '%d/%m/%y')
        self._inmueble = inmueble
        self._inquilino = inquilino
        self._estado = estado
        Contrato.lista_contratos.append(self)

    # Getters and Setters

    def getId(self):
        return self._id

    def setId(self, id):
        self._id = id

    def getClausula(self):
        return self._clausula

    def setClausula(self, clausula):
        self._clausula = clausula

    def getFechaInicio(self):
        return self._fechaInicio

    def setFechaInicio(self, fechaInicio):
        self._fechaInicio = dt.datetime.strptime(fechaInicio, '%d/%m/%y')

    def getFechaFin(self):
        return self._fechaFin

    def setFechaFin(self, fechaFin):
        self._fechaFin = dt.datetime.strptime(fechaFin, '%d/%m/%y')

    def getInmueble(self):
        return self._inmueble

    def setInmueble(self, inmueble):
        self._inmueble = inmueble

    def getInquilino(self):
        return self._inquilino

    def setInquilino(self, inquilino):
        self._inquilino = inquilino

    def isEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado

    # Fin Getters and Setters

    def renovarContrato(self):  # Aqui tomamos la fecha de finalización del contrato y le sumamos 30 días.
        self._fechaFin += 30    # Toca revisar cómo trabajar con fechas

    def notificarCierreContrato(self):  #Este metodo se encarga de cerrar el contrato en caso de que la fecha fin sea igual a la fecha actual más 1 dias y notifique de esto al usuario.
        if self.getFechaFin() == (dt.datetime.now() + dt.timedelta(days=1)): # Se necesita buscar respecto a Datetime en Python
           self.setEstado(not self.isEstado())
           notificacion = "Se ha cerrado el contrato"
           return notificacion
        else:
            notificacion = "Aun queda tiempo"
            return notificacion

    def inhabilitarContrato(self):  # Este método se encarga de evaluar si el estado del contrato es true y en este se cambia a false. En caso de que esté en false no se hace nada.
        if(self.isEstado() == True):
            self.setEstado(not self.isEstado())

    def mostrarContrato(self):
        return "Contrato {" + "id = " + str(self.getId()) + "\n, clausula = " + self.getClausula() + "\n fecha inicio = " + str(self.getFechaInicio()) + ", fecha fin = " + str(self.getFechaFin()) + "\n, inmueble = " + str(self.getInmueble()) + "\n, inquilino = " + self.getInquilino() + "\n, estado = " + str(self.isEstado()) + "}"
